swarm/network/protocol.go

```diff
@@ -39,12 +39,26 @@ import (
 
 	"github.com/ethereum/go-ethereum/contracts/chequebook"
 	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/metrics"
 	"github.com/ethereum/go-ethereum/p2p"
 	bzzswap "github.com/ethereum/go-ethereum/swarm/services/swap"
 	"github.com/ethereum/go-ethereum/swarm/services/swap/swap"
 	"github.com/ethereum/go-ethereum/swarm/storage"
 )
 
+//metrics variables
+var (
+	storeRequestMsgCounter    = metrics.NewCounter("network.protocol.msg.storerequest.count")
+	retrieveRequestMsgCounter = metrics.NewCounter("network.protocol.msg.retrieverequest.count")
+	peersMsgCounter           = metrics.NewCounter("network.protocol.msg.peers.count")
+	syncRequestMsgCounter     = metrics.NewCounter("network.protocol.msg.syncrequest.count")
+	unsyncedKeysMsgCounter    = metrics.NewCounter("network.protocol.msg.unsyncedkeys.count")
+	deliverRequestMsgCounter  = metrics.NewCounter("network.protocol.msg.deliverrequest.count")
+	paymentMsgCounter         = metrics.NewCounter("network.protocol.msg.payment.count")
+	invalidMsgCounter         = metrics.NewCounter("network.protocol.msg.invalid.count")
+	handleStatusMsgCounter    = metrics.NewCounter("network.protocol.msg.handlestatus.count")
```

Comment:
I feel like these should just be one metric `network.protocol.msg.count` with a `type` tag.

Reply:
The current API for `go-ethereum/metrics` doesn't allow additional tags per metric, as far as I know. @nonsense can you confirm?

Reply:
Well,am not too sure about that but I think you are right.And please explain more of this tag per metric.

Reply:
The `go-metrics` API indeed has no support for fields, or tags on individual measurements (counters, gauges, timers, etc.). When using `go-metrics` tags can be added per reporter - the functionality which periodically emits the registry to the backend.

Some `statsd` clients support tags on a per measurement level (for example https://godoc.org/github.com/ooyala/go-dogstatsd) although most also resort to adding tags on a per reporter/registry level.

Reply:
Is this something we could easily add to go-metrics? If not, then let's not worry about it.

Reply:
No, AFAIK we cannot easily add it to `go-metrics`, we'd have to change most of the existing API.